### ISSUE TITLE
S1450 "Private fields only used as local variables in methods should become local variables" not triggered by rule sample

### ISSUE THREAD

**Issue by ohadschn** [SonarSource/sonar-csharp#1536]
### Description

The sample non-compliant code from the rule's description does not trigger the rule.

### Repro steps

```cs
public class Foo
{
    private int _singularField;
    public void DoSomething(int x)
    {
        _singularField = x + 5;
        if (_singularField == 0) { /* ... */ }
    }
}
```

### Expected behavior

S1450 should be triggered

### Actual behavior

Build succeeds with zero warnings and zero errors.

### Related information

* SonarC# Version: 7.3.0.5690 (latest at the time of writing)
* Visual Studio 2017 (15.7.4)

**Comment by valhristov** [comment]
Hi @ohadschn I cannot reproduce the problem, the rule raises on the snippet:

![image](https://user-images.githubusercontent.com/23195959/43462132-928fe2b0-94d5-11e8-8d99-754d7d22af57.png)

What's the name of your project? S1450 does not run on projects that contain "test" in their name and that's very often the reason for false negatives with some rules...

**Comment by ohadschn** [comment]
I honestly don't remember, this was almost a month ago.
At any rate, I think the rule's description should mention this test exclusion behavior...

**Comment by valhristov** [comment]
Anyway, with the next version of SonarC# we will be releasing a brand new implementation of this rule that does not rely on recompilation and should be much more stable detecting issues.
